packages/opentelemetry-core/src/metrics/NoopMeter.ts

```diff
@@ -0,0 +1,150 @@
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  CounterHandle,
+  DistributedContext,
+  GaugeHandle,
+  Meter,
+  Metric,
+  MetricOptions,
+  MeasureHandle,
+  SpanContext,
+} from '@opentelemetry/types';
+
+/**
+ * NoopMeter is a noop implementation of the {@link Meter} interface. It reuses constant
+ * NoopMetrics for all of its methods.
+ */
+export class NoopMeter implements Meter {
+  constructor() {}
+
+  /**
+   * Returns constant noop measure.
+   * @param name the name of the metric.
+   * @param [options] the metric options.
+   */
+  public createMeasure(
+    name: string,
+    options?: MetricOptions
+  ): Metric<MeasureHandle> {
+    return NOOP_MEASURE_METRIC;
+  }
+
+  /**
+   * Returns a constant noop counter.
+   * @param name the name of the metric.
+   * @param [options] the metric options.
+   */
+  createCounter(name: string, options?: MetricOptions): Metric<CounterHandle> {
+    return NOOP_COUNTER_METRIC;
+  }
+
+  /**
+   * Returns a constant gauge metric.
+   * @param name the name of the metric.
+   * @param [options] the metric options.
+   */
+  createGauge(name: string, options?: MetricOptions): Metric<GaugeHandle> {
+    return NOOP_GAUGE_METRIC;
+  }
+}
+
+export class NoopMetric<T> implements Metric {
+  private readonly _handle: T;
+
+  constructor(handle: T) {
+    this._handle = handle;
+  }
+  /**
+   * Returns a Handle associated with specified label values.
+   * It is recommended to keep a reference to the Handle instead of always
+   * calling this method for every operations.
+   * @param labelValues the list of label values.
+   */
+  getHandle(labelValues: string[]): T {
+    return this._handle;
+  }
+
+  /**
+   * Returns a Handle for a metric with all labels not set.
+   */
+  getDefaultHandle(): T {
+    return this._handle;
+  }
+
+  /**
+   * Removes the Handle from the metric, if it is present.
+   * @param labelValues the list of label values.
+   */
+  removeHandle(labelValues: string[]): void {
+    return;
+  }
+
+  /**
+   * Clears all timeseries from the Metric.
+   */
+  clear(): void {
+    return;
+  }
+
+  setCallback(fn: () => void): void {
+    return;
+  }
+}
+
+export class NoopCounterHandle implements CounterHandle {
+  add(value: number): void {
+    return;
+  }
+}
+
+export class NoopGaugeHandle implements GaugeHandle {
+  set(value: number): void {
+    return;
+  }
+}
+
+export class NoopMeasureHandle implements MeasureHandle {
+  record(value: number): void {
+    return;
+  }
+
+  record(value: number, distContext: DistributedContext): void {
```

Comment:
Duplicate function implementation is not allowed, I think you should change this to: 

```
  record(
    value: number,
    distContext?: DistributedContext,
    spanContext?: SpanContext
  ): void {
    return;
  }
```

Reply:
or you can use typescript overload like something : 
https://www.typescriptlang.org/play/index.html#code/ATDGBsEMGduA5A9ogDsA3gKBCATgU1EVwBMAKAN0nAFd8AuYAOxoFsAjfXAGmBIEtoAFwDCiJkPwAPIY0hMAngEpGFRPxIBubDgJFSZHThxVaDZm048jxvoNHjJMuYu43j0FPLETpQgPwuCjYqwGoaGO66+EI0uEzaxgC+OikgQA